File: SYP.Web/Modules/ServerTypes/Setting/CurrencyListEditorColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CurrencyListRow } from "./CurrencyListRow";

export interface CurrencyListEditorColumns {
    Id: Column<CurrencyListRow>;
    Code: Column<CurrencyListRow>;
    Name: Column<CurrencyListRow>;
    CodeType: Column<CurrencyListRow>;
    Symbol: Column<CurrencyListRow>;
    IsActive: Column<CurrencyListRow>;
    InsertUserId: Column<CurrencyListRow>;
    InsertDate: Column<CurrencyListRow>;
    UpdateUserId: Column<CurrencyListRow>;
    UpdateDate: Column<CurrencyListRow>;
    IsDefaultCurrency: Column<CurrencyListRow>;
    DefaultExchangeType: Column<CurrencyListRow>;
    TenantId: Column<CurrencyListRow>;
}

export class CurrencyListEditorColumns extends ColumnsBase<CurrencyListRow> {
    static readonly columnsKey = 'Setting.CurrencyListEditor';
    static readonly Fields = fieldsProxy<CurrencyListEditorColumns>();
}

File: SYP.Web/Modules/Setting/CurrencyList/Editor/CurrencyListEditorColumns.cs
namespace SYP.Setting.Columns;

[ColumnsScript("Setting.CurrencyListEditor")]
[BasedOnRow(typeof(CurrencyListRow), CheckNames = true)]
public class CurrencyListEditorColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Code { get; set; }
    public string Name { get; set; }
    public short CodeType { get; set; }
    public string Symbol { get; set; }
    public bool IsActive { get; set; }
    public int InsertUserId { get; set; }
    public DateTime InsertDate { get; set; }
    public int UpdateUserId { get; set; }
    public DateTime UpdateDate { get; set; }
    public bool IsDefaultCurrency { get; set; }
    public short DefaultExchangeType { get; set; }
    public int TenantId { get; set; }
}